app/javascript/angular/code/controllers/_mobile_sessions_map_ctrl.js

```diff
@@ -141,6 +131,14 @@ export const MobileSessionsMapCtrl = (
         sessionsUtils.updateCrowdMapLayer($scope.sessions.allSessionIds());
       });
 
+      elmApp.ports.findLocation.subscribe(location => {
+        FiltersUtils.findLocation(location, params, map);
+      });
+
+      map.onPanOrZoom(() => {
+        FiltersUtils.clearLocation(elmApp, params);
```

Comment:
Have you thought keeping the port action on this side? Passing the entire elm app to a util function seems like delegating too much responsibility